go.mod

```diff
@@ -0,0 +1,49 @@
+module github.com/dask/dask-kubernetes/v2023
```

Comment:
I don't think my skills are a lot better here, I'm probably just a few hours of googling ahead :-D 

This is something that I had to do to allow for a major version != 0 or 1. Without this, a dependent module would not be able to specify say: 
```
require(
    github.com/dask/dask-kubernetes v2023.3.2
) 
```
as this would fail with `version "v2.5.10" invalid: should be v0 or v1, not v2023 syntax`. After finding https://github.com/golang/go/issues/35732 and reading the associated [docs](https://go.dev/doc/modules/release-workflow#breaking), I've realized that each breaking change (for >=2.0.0) is realized in the form of a new module, postfixed with the major version (as here). 

I'm not a particular fan of this given that the `dask` ecosystem uses CalVer, but `go` is built around SemVer, so I don't think there's much we can do if we want to have things in the same repo. It also means that we'll have to increase this major version each year, which doesn't seem ideal.

Happy to reconsider whether we should move the `go` code here in case you have concerns

Reply:
Ok thanks for the clarification. I never knew that about versions >1. My [thoughts on CalVer are well documented](https://jacobtomlinson.dev/posts/2023/sometimes-i-regret-using-calver/), but it's the situation we are in. 

I wonder if we could add some automation to alert us when this gets out of sync. I can totally imagine myself tagging `2024.1.0` and forgetting to update this.

Reply:
Interesting post! I've been following the github discussion around why `dask` switched to CalVer, so that's great insight in how it's turned out. The user perspective in that post around communicating breaking changes definitely resonates.

Instead of an automation that alerts us, I've built [this automation script](https://github.com/bstadlbauer/dask-kubernetes/blob/add-operator-go-client/.github/workflows/update-go-package-version.yml) that updates the version automatically each year. Happy to switch to a simple reminder in case you don't want the added complexity.